loony_dev/agents/coding.py

```
@@ -64,6 +65,270 @@ def execute(self, task: Task) -> TaskResult:
+            except GitError as exc:
+                logger.warning("Issue #%d: git error during commit/push: %s", task.issue.number, exc)
+                break
+
+        if not commit_succeeded or task.review_exhausted:
+            if not commit_succeeded:
+                task.mark_commit_exhausted(hook_failed_output)
+                wip_msg = f"[WIP] {commit_msg}"
+                logger.warning("Issue #%d: committing as WIP: %s", task.issue.number, wip_msg)
+                try:
+                    git.commit_and_push(wip_msg, branch)
+                except Exception as exc:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)